.github/workflows/ci.yml

```diff
@@ -71,6 +71,34 @@ jobs:
         chmod a+x builder
         ./builder build -p ${{ env.PACKAGE_NAME }}
 
+  # Test that everything works with Swift 6 strict concurrency. We can remove this CI in the future once we raise our minimum Swift version to Swift 6.
+  macos-swift6:
+    runs-on: macos-15
+    env:
+      DEVELOPER_DIR: /Applications/Xcode.app
+      XCODE_DESTINATION: 'OS X'
+      NSUnbufferedIO: YES
+    strategy:
+      fail-fast: false
+    steps:
+    - uses: aws-actions/configure-aws-credentials@v4
+      with:
+        role-to-assume: ${{ env.CRT_CI_ROLE }}
+        aws-region: ${{ env.AWS_DEFAULT_REGION }} 
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        submodules: true
+        fetch-depth: 0
+    - name: Set Swift version to 6
+      run: |
+        sed -i '' 's/swift-tools-version:5\.7/swift-tools-version:6\.0/' Package.swift
+        cat Package.swift | head -3
```

Comment:
if `Package.swift` doesn't have the exact string `swift-tools-version:5.7`, this step will just fail silently, which will make it hard to debug and probably leave the CI doesn't do the expected thing for us.


GenAI suggested something like:
```
Solution 1: Combined sed command with conditional exit
Bash
 Copy code
sed -i '' '/swift-tools-version:5\.7/{s/swift-tools-version:5\.7/swift-tools-version:6\.0/;b};q1' Package.swift
This will:
Search for the pattern
If found, make the substitution and branch to end of script (with b)
If not found, exit with code 1 (with q1)
```

Also, using the exact version seems less future proven. I am not sure if we will ever change the version in the package.swift, but in general, maybe just match all version instead.

Reply:
Thanks, I have made it more generic so that it works with any 5.xx version and also fails it can't replace it.